### ISSUE TITLE
xmlrpc_encode_request

### ISSUE THREAD

**Issue by parmentelat** [gggeek/phpxmlrpc#105]
hiya lads

I use this library, in conjunction with `xmlrpc_encode_request()`

I just upgraded a system that was working under fedora 35 (that comes with php-8.0) to fedora37 (that has php-8.1) and it now crashes, with the following error
```
Fatal error: Uncaught Error: Call to undefined function xmlrpc_encode_request() in /usr/share/plc_api/php/plc_api.php:
```

I did notice that [the official documentation for `xmlrpc_encode_request()` here](https://www.php.net/manual/en/function.xmlrpc-encode-request.php) states it is experimental (must have been so for 20 years, but that's another story altogether....)

so my question is: is it possible that this function indeed has been removed between 8.0 and 8.1 ?
also I expect this library offers tools to work around this, does it ?

thank you !

**Comment by parmentelat** [comment]
hiya again

sorry, my problem is totally not relevant to phpxmlrpc, it's entirely on my end
apologies for the noise, let me close this

**Comment by gggeek** [comment]
Hi there.

It seems that your issue is quite similar to something that happened already in 2017 - I found a ticket from yourself at https://github.com/gggeek/phpxmlrpc/issues/45 :-)

In case your proficiency with PHP and PHPXMLRPC has not increased considerably since then, here's a recap of the situation, and a proposal for a solution:

- the function `xmlrpc_encode_request` was never part of this library. It comes from a php extension, ie. some code written in C. The extension is named more or less 'xmlrpc for php'. This library is named 'phpxmlrpc' 
- the extension has been long unmantained. It has been removed from official php builds since php 8.0 (see f.e. https://php.watch/versions/8.0/xmlrpc). It can still be found in PECL (at https://pecl.php.net/package/xmlrpc), and thus can be installed by compiling it, in case you really need it and your php setup includes compiling php extensions. Its maintainer strongly discourages anyone from using it, though
- I have provided for a while an alternative implementation of the functions which are part of the xmlrpc extension. My implementation is written in php, so you don't need to compile anything to use it. That code was initially part of the package https://github.com/gggeek/phpxmlrpc-extras, but it has since been moved to https://github.com/gggeek/polyfill-xmlrpc

Long story short: in order to be able to use the function `xmlrpc_encode_request`, the recommended way nowadays is to use the gggeek/polyfill-xmlrpc package.
The recommended way to do that is by using Composer.

**Comment by parmentelat** [comment]
hey Gaetano
thanks for the reply
as I said this was a mistake on my end; as you outlined I was just running again into the same issue as a few years back...

turns out that while upgrading to fedora-37 I was messing with the packaging of my app, which should have included your library, but the packaging was faulty and your library was not in there; so my php code was running the same old branch as in 2017, hence the duplicated issue

sorry about that; as you have noticed my php fluency is very low, and I guess I need a second brain to remember all the errors of the past ;-)
